### ISSUE TITLE
Switch to golang/dep for dependencies

### ISSUE THREAD

**Issue by prydonius** [kubernetes-helm/monocular#354]
Switch to the official https://github.com/golang/dep tool for dependency management

**Comment by slintes** [comment]
Hi, I want to get familiar with monocular and also with the new `dep` dependency manager, so I would like to pick up this issue. I see that you use the `deis/go-dev` docker image for go tooling, currently version `v0.22.0`, which includes `go` version `1.7.3`.
`dep` was added in `deis/go-dev:v0.24.0`, which includes `go 1.8.1`.
But because `dep` is still under heavy development, it probably makes sense to even use the latest `deis/go-dev:v1.4.0` image, which includes `go 1.9`.

Long story short: are there reasons to not use `go 1.9`?

https://github.com/deis/docker-go-dev/releases

**Comment by prydonius** [comment]
@slintes as discussed on Slack feel free to take this :).

Sorry forgot to answer your question about 1.9, no particular reason, feel free to update to the latest go-dev image!
